src/lib/core/data-table/data-table.ts

```diff
@@ -142,40 +175,54 @@ export class CdkTable<T> implements CollectionViewer {
       this._columnDefinitionsByName.set(columnDef.name, columnDef);
     });
 
-    // Get and merge the streams for column changes made to the row defs
-    const rowDefs = [...this._rowDefinitions.toArray(), this._headerDefinition];
-    const columnChangeStreams =
-        rowDefs.map((rowDef: BaseRowDef) => rowDef.columnsChange);
-    this._columnsChange = Observable.merge(...columnChangeStreams);
+    // Re-render the rows if any of their columns change.
+    // TODO(andrewseguin): Determine how to only re-render the rows that have their columns changed.
+    const rowColumnsChange =
+        this._rowDefinitions.toArray().map((rowDef: BaseRowDef) => rowDef.columnsChange);
```

Comment:
QueryList has a `map` function so you don't need `toArray`
https://angular.io/api/core/QueryList

You also shouldn't need a type in the map function since it is given by the generic